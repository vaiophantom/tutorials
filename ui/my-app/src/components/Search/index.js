import { useState } from "react";
import {
  Form,
  InputGroup,
  DropdownButton,
  Dropdown,
  Image,
} from "react-bootstrap";
import IMAGES from "../../assets/images";
import CONSTANT from "../../constants";

const Search = (props) => {
  const { handleSearch } = props;
  const [typeSearch, setTypeSearch] = useState(CONSTANT.TYPE_SEARCH[0]);
  const [valueSearch, setValueSearch] = useState("");

  const onChangeSearch = (e) => {
    const { value } = e.target;
    setValueSearch(value);
  };

  const onClickSearch = () => {
    handleSearch(typeSearch, valueSearch);
  };

  return (
    <>
      <InputGroup style={{ backgroundColor: "white", alignItems: "center" }}>
        <DropdownButton
          defaultValue={"asdasd"}
          variant="outline-primary"
          title={typeSearch}
        >
          {CONSTANT.TYPE_SEARCH.map((item, idx) => {
            return (
              <Dropdown.Item
                key={idx}
                onClick={() => {
                  setTypeSearch(item);
                }}
              >
                {item}
              </Dropdown.Item>
            );
          })}
        </DropdownButton>
        <Form.Control
          value={valueSearch}
          onChange={onChangeSearch}
          placeholder="Search products based on category name, product name, price, branch..."
          aria-describedby="basic-addon3"
        />
        <Image
          className=""
          onClick={onClickSearch}
          src={IMAGES.search}
          height={30}
          width={30}
        />
      </InputGroup>
    </>
  );
};
export default Search;
