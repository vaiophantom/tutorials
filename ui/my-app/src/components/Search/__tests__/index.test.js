import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Search from '../';

it('renders Search', async () => {
 const {container} = render(<Search />, {wrapper: BrowserRouter});
 expect(container).toMatchSnapshot();
});
