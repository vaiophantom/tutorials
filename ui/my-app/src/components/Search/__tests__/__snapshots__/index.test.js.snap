// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Search 1`] = `
<div>
  <div
    class="input-group"
    style="background-color: white; align-items: center;"
  >
    <button
      aria-expanded="false"
      class="dropdown-toggle btn btn-outline-primary"
      type="button"
    >
      name
    </button>
    <div
      aria-labelledby=""
      class="dropdown-menu"
      x-placement="bottom-start"
    >
      <a
        class="dropdown-item"
        data-rr-ui-dropdown-item=""
        href="#"
        role="button"
        tabindex="0"
      >
        name
      </a>
      <a
        class="dropdown-item"
        data-rr-ui-dropdown-item=""
        href="#"
        role="button"
        tabindex="0"
      >
        branch
      </a>
      <a
        class="dropdown-item"
        data-rr-ui-dropdown-item=""
        href="#"
        role="button"
        tabindex="0"
      >
        category
      </a>
      <a
        class="dropdown-item"
        data-rr-ui-dropdown-item=""
        href="#"
        role="button"
        tabindex="0"
      >
        color
      </a>
    </div>
    <input
      aria-describedby="basic-addon3"
      class="form-control"
      placeholder="Search products based on category name, product name, price, branch..."
      value=""
    />
    <img
      class=""
      height="30"
      src="search.jpeg"
      width="30"
    />
  </div>
</div>
`;
